File: test/samples/floats/floats.jl
import Base: parse

"""
Represents coordinates
"""
struct Coordinates
    """X"""
    x::Float64
    """Y"""
    y::Float64
    """Z"""
    z::Float64
end


"""
Mix of fields that go on one line
"""
struct InlinedMix
    """an integer"""
    integer::Int
    """a char"""
    char::Char
    """a float"""
    float::Float64
end


"""
a struct of chars
"""
struct MultilineMix
    """an other integer"""
    integer_2::Int
    """a string of size 5"""
    string::String
    """an other float"""
    float_2::Float64
end


"""
- `f::Float64`: a float
- `g::Float64`: a float, greater than f
- `point::Coordinates`: some coordinates
- `n::Int`: a number
- `float_list::Vector{Float64}`: a list of floats
- `other_list::Vector{Float64}`: a list of floats
- `inlined::Vector{InlinedMix}`: some inlined structs
- `multiline::MultilineMix`: a multiline struct
"""
function floats(f::Float64, g::Float64, point::Coordinates, n::Int, float_list::Vector{Float64}, other_list::Vector{Float64}, inlined::Vector{InlinedMix}, multiline::MultilineMix)
    # TODO Parsing is often easy, reprint mode is harder
end

function parse(::Type{Coordinates}, s::AbstractString)
    s = split(s)
    Coordinates(parse(Float64, s[1]), parse(Float64, s[2]), parse(Float64, s[3]))
end

function parse(::Type{InlinedMix}, s::AbstractString)
    s = split(s)
    InlinedMix(parse(Int, s[1]), s[2][1], parse(Float64, s[3]))
end

function read_struct_multiline_mix()
    integer_2 = parse(Int, readline())
    string = readline()
    float_2 = parse(Float64, readline())
    MultilineMix(integer_2, string, float_2)
end

f = parse(Float64, readline())
g = parse(Float64, readline())
point = parse(Coordinates, readline())
n = parse(Int, readline())
float_list = map(s -> parse(Float64, s), split(readline()))
other_list = map(s -> parse(Float64, s), split(readline()))
inlined = [ parse(InlinedMix, readline()) for _=1:3 ]
multiline = read_struct_multiline_mix()
floats(f, g, point, n, float_list, other_list, inlined, multiline)

File: test/samples/floats/floats.cs
using System;
using System.Globalization;

/// Represents coordinates
struct Coordinates
{
    public double x; //!< X
    public double y; //!< Y
    public double z; //!< Z
}

/// Mix of fields that go on one line
struct InlinedMix
{
    public int integer; //!< an integer
    public char @char; //!< a char
    public double @float; //!< a float
}

/// a struct of chars
struct MultilineMix
{
    public int integer2; //!< an other integer
    public string @string; //!< a string of size 5
    public double float2; //!< an other float
}

class Program
{
    /// \param f a float
    /// \param g a float, greater than f
    /// \param point some coordinates
    /// \param n a number
    /// \param floatList a list of floats
    /// \param otherList a list of floats
    /// \param inlined some inlined structs
    /// \param multiline a multiline struct
    static void Floats(double f, double g, Coordinates point, int n, double[] floatList, double[] otherList, InlinedMix[] inlined, MultilineMix multiline)
    {
        /* TODO Parsing is often easy, reprint mode is harder */
    }

    static void Main()
    {
        double f = double.Parse(Console.ReadLine(), CultureInfo.InvariantCulture);
        double g = double.Parse(Console.ReadLine(), CultureInfo.InvariantCulture);
        string[] words = Console.ReadLine().Split(' ');
        Coordinates point = new Coordinates {x = double.Parse(words[0], CultureInfo.InvariantCulture), y = double.Parse(words[1], CultureInfo.InvariantCulture), z = double.Parse(words[2], CultureInfo.InvariantCulture)};
        int n = int.Parse(Console.ReadLine());
        double[] floatList = Array.ConvertAll(Console.ReadLine().Split(new char[] {' '}, StringSplitOptions.RemoveEmptyEntries), x => double.Parse(x, CultureInfo.InvariantCulture));
        double[] otherList = Array.ConvertAll(Console.ReadLine().Split(' '), x => double.Parse(x, CultureInfo.InvariantCulture));
        InlinedMix[] inlined = new InlinedMix[3];
        for (int i = 0; i < 3; ++i)
        {
            string[] words1 = Console.ReadLine().Split(' ');
            inlined[i] = new InlinedMix {integer = int.Parse(words1[0]), @char = words1[1][0], @float = double.Parse(words1[2], CultureInfo.InvariantCulture)};
        }
        MultilineMix multiline;
        multiline.integer2 = int.Parse(Console.ReadLine());
        multiline.@string = Console.ReadLine();
        multiline.float2 = double.Parse(Console.ReadLine(), CultureInfo.InvariantCulture);

        Floats(f, g, point, n, floatList, otherList, inlined, multiline);
    }
}
